hooks/useHotWalletPluginTokenAccounts.ts

```diff
@@ -0,0 +1,112 @@
+import { MintInfo, u64 } from '@solana/spl-token'
+import BigNumber from 'bignumber.js'
+import { PublicKey } from '@solana/web3.js'
+import {
+  getOwnedTokenAccounts,
+  TokenProgramAccount,
+  tryGetMint,
+} from '@utils/tokens'
+import { useCallback, useEffect, useState } from 'react'
+import useWalletStore from 'stores/useWalletStore'
+import tokenService from '@utils/services/token'
+import { BN } from '@project-serum/anchor'
+import { HotWalletAccount } from './useHotWallet'
+import { getSplTokenNameByMint } from '@utils/splTokens'
+
+export type HotWalletTokenAccounts = {
+  publicKey: PublicKey
+  mint: PublicKey
+  decimals: number
+  amount: u64
+  mintName?: string
+  usdMintValue: number
+  usdTotalValue: u64
+}[]
+
+const useHotWalletPluginTokenAccounts = (
+  hotWalletAccount: HotWalletAccount
+) => {
+  const connection = useWalletStore((store) => store.connection)
+  const [
+    tokenAccounts,
+    setTokenAccounts,
+  ] = useState<HotWalletTokenAccounts | null>(null)
+
+  const loadTokenAccounts = useCallback(async () => {
+    if (!connection.current) return
+
+    const ownedTokenAccounts = await getOwnedTokenAccounts(
+      connection.current,
+      hotWalletAccount.publicKey
+    )
+
+    const tokenMintAddresses = [
+      ...new Set(ownedTokenAccounts.map(({ account: { mint } }) => mint)),
+    ]
+
+    const mintInfos = (
+      await Promise.all(
+        tokenMintAddresses.map((tokenMintAddress) =>
+          tryGetMint(connection.current, tokenMintAddress)
+        )
+      )
+    ).reduce(
+      (acc, mintInfo) => {
+        if (!mintInfo) throw new Error('Cannot load mint info')
+
+        acc[mintInfo.publicKey.toBase58()] = {
+          ...mintInfo,
+          name: getSplTokenNameByMint(mintInfo.publicKey),
+          usdValue: tokenService.getUSDTokenPrice(
+            mintInfo.publicKey.toBase58()
+          ),
+        }
+
+        return acc
+      },
+      {} as {
+        [key: string]: TokenProgramAccount<MintInfo> & {
+          name?: string
+          usdValue: number
+        }
+      }
+    )
+
+    setTokenAccounts(
+      ownedTokenAccounts
+        .map((tokenAccount) => {
+          const mintInfo = mintInfos[tokenAccount.account.mint.toBase58()]
+
+          return {
+            mint: tokenAccount.account.mint,
+            publicKey: tokenAccount.publicKey,
+            amount: tokenAccount.account.amount,
+            decimals: mintInfo.account.decimals,
+            mintName: mintInfo.name,
+            usdMintValue: mintInfo.usdValue,
+            usdTotalValue: new BN(
+              new BigNumber(tokenAccount.account.amount.toString())
+                .multipliedBy(mintInfo.usdValue)
+                .integerValue()
+                .toString()
+            ),
+          }
+        })
+        .sort((a, b) => b.amount.toNumber() - a.amount.toNumber())
+    )
+  }, [
+    connection,
+    JSON.stringify(tokenService._tokenPriceToUSDlist),
```

Comment:
lol

Reply:
yeah I took that from an other part of the governance ^^' We should really refactor all of this:

- List of tokens and theirs logo etc.
- Actual token price

Put it inside of a service and make it available through a store.